src/package-tree.ts

```diff
@@ -1,53 +1,110 @@
-interface PointOfInterest {
+import * as fs from 'fs';
+import * as path from 'path';
+import pify from 'pify';
+
+export interface PointOfInterest {
   type: string;
   filename: string;
   lineNum: number;
 }
 
-export interface PackageTree {
+export interface PackageTree<T> {
   rootPackageName: string;
   version: string;
-  data: PointOfInterest[];
-  dependencies: PackageTree[];
+  data: T;
+  dependencies: Array<PackageTree<T>>;
 }
 
-function getDependencies(pjson: string): PackageTree {
-  // todo -- this function will probably be recursive
-  throw new Error('not implemented');
-  // compute result
-  //   let result: PackageTree;
-  //   return result;
+export const readFilep = pify(fs.readFile);
+export interface ReadFileP {
+  (path: string, encoding: string): Promise<string>;
 }
 
-function getPOIforPackageTree(packageTree: PackageTree): PackageTree {
-  // TODO:
-  throw new Error('not implemented');
-  // step 1: get POI modules for current package
-  // step 2: add POI to PackageTree Object
-}
+/**
+ * Takes in the contents of a project's package.json file as a string and returns a PackageTree ob
+ * @param pjson is the contents of a project's package.json file as a string
+ */
+export async function generatePackageTree(rootDir: string, customReadFilep: ReadFileP): Promise<PackageTree<undefined>> {
+  // todo -- this function will probably be recursive
+  customReadFilep = customReadFilep || readFilep;
+  // step 0: read in package.json and package-lock.json
+  try{
```

Comment:
Run clang format. Needs to be a space after the `try`, and the this entire block needs to be indented.